### PR TITLE
[TypeScript][VA & Skill] Add Skill connection in the VA's Functional Test

### PR DESCRIPTION
**Pull request by DiegoCardozo94**
**This PR depends on 3463 and 3542**
Related to 3290

### Purpose
*What is the context of this pull request? Why is it being done?*
In the TypeScript VA's Functional Test was missing the connection between a Virtual Assistant and Skill in TypeScript.

### Changes
*Are there any changes that need to be called out as significant or particularly difficult to grasp? (Include illustrative screenshots for context if applicable.)*

- Copy and adapt C# Virtual Assistant & Skill Functional Test as the responses' length between C# and TS are different
- Add `YAML` file for the TypeScript Virtual Assistant that contains Skill configuration
- Connect Virtual Assistant with a Skill, both in TypeScript
- Update Virtual Assistant documentation

Sample Assistant YAML file
image

Pipeline running successfully
image

### Tests
*Is this covered by existing tests or new ones? If no, why not?*
Adding the functional test for the Virtual Assistant & Skill. Also, tested it locally/manually.

### Feature Plan
*Are there any remaining steps or dependencies before this issue can be fully resolved? If so, describe and link to any relevant pull requests or issues.*
*-*

#### General
- [ ] I have commented my code, particularly in hard-to-understand areas	
- [ ] I have added or updated the appropriate tests	
- [X] I have updated related documentation

### CODE REVIEW

File: templates/typescript/samples/sample-assistant/functional-tests/sample-assistant.yml [hunk 1]
```diff
@@ -267,77 +384,171 @@ jobs:
...
+     Write-Host "Publish log"
+     Write-Host "===="
+     Get-Content -Path $(Join-Path "Deployment" "publish_log.txt")
+     Write-Host "`r`n"
+
     errorActionPreference: continue
-    workingDirectory: 'templates/typescript/samples/$(BotName)'
-    displayName: 'Show log contents'
+    workingDirectory: 'templates/typescript/samples/functionalTests/$(SkillBotName)'
+    displayName: 'Debug: Skill - Show log contents copy'
```
Comment (Batta32) [inline templates/typescript/samples/sample-assistant/functional-tests/sample-assistant.yml:548]: We should delete the `copy` keyword

File: templates/typescript/samples/sample-assistant/functional-tests/sample-assistant.yml [hunk 2]
```diff
@@ -267,77 +384,171 @@ jobs:
...
+     Write-Host "`r`n"
+    
+     echo "##vso[task.setvariable variable=BOTID;]$(SkillBotWebAppName)"
+    
+    workingDirectory: 'templates/typescript/samples/functionalTests/$(SkillBotName)'
+    displayName: 'Test: Skill - Get channel secrets'
 
   - task: DotNetCoreCLI@2
-    displayName: 'Run dotnet restore'
+    displayName: 'Test: Skill -  Execute functional tests'
```
Comment (Batta32) [inline templates/typescript/samples/sample-assistant/functional-tests/sample-assistant.yml:438]: Delete the redundant whitespace

File: templates/typescript/samples/sample-assistant/functional-tests/sample-assistant.yml [hunk 3]
```diff
@@ -267,77 +384,171 @@ jobs:
...
-    displayName: 'Run dotnet test on functional tests'
-    inputs:
-      command: test
-      projects: 'templates/typescript/samples/sample-assistant/functional-tests/VirtualAssistantSample.FunctionalTests/VirtualAssistantSample.FunctionalTests.csproj'
-      arguments: '--configuration $(BuildConfiguration) --filter TestCategory=FunctionalTests'
-      publishTestResults: false
+  - pwsh: '.\Deployment\Scripts\publish.ps1 -name "$(VirtualAssistantBotWebAppName)" -resourceGroup "$(VirtualAssistantBotResourceGroup)"'
+    errorActionPreference: continue
+    workingDirectory: 'templates/typescript/samples/functionalTests/$(VirtualAssistantBotName)'
+    displayName: 'Test: VA - Publish with connected Skill'
```
Comment (Batta32) [inline templates/typescript/samples/sample-assistant/functional-tests/sample-assistant.yml:470]:
```suggestion
    displayName: 'Test: VA - Publish with connected Skill'
```

File: templates/typescript/samples/sample-assistant/functional-tests/sample-assistant.yml [hunk 4]
```diff
@@ -267,77 +384,171 @@ jobs:
...
+     Write-Host "DirectLine Bot Name: $(VirtualAssistantBotWebAppName)"
+     Write-Host "`r`n"
+    
+     echo "##vso[task.setvariable variable=BOTID;]$(VirtualAssistantBotWebAppName)"
+    
+    workingDirectory: 'templates/typescript/samples/functionalTests/$(VirtualAssistantBotName)'
+    displayName: 'Test: VA - Get channel secrets'
+
+  - task: DotNetCoreCLI@2
+    displayName: 'Test: VA -  Execute functional tests'
```
Comment (Batta32) [inline templates/typescript/samples/sample-assistant/functional-tests/sample-assistant.yml:503]: Delete the redundant whitespace

File: templates/typescript/samples/sample-assistant/functional-tests/sample-assistant.yml [hunk 5]
```diff
@@ -267,77 +384,171 @@ jobs:
...
+    inputs:
+      command: test
+      projects: 'templates/typescript/samples/sample-assistant/functional-tests/VirtualAssistantSample.FunctionalTests/VirtualAssistantSample.FunctionalTests.csproj'
+      arguments: '-v n  --configuration $(BuildConfiguration) --filter TestCategory=FunctionalTests'
```
Comment (Batta32) [inline templates/typescript/samples/sample-assistant/functional-tests/sample-assistant.yml:507]:
```suggestion
      arguments: '-v n --configuration $(BuildConfiguration) --filter TestCategory=FunctionalTests'
```

File: templates/typescript/samples/sample-assistant/functional-tests/sample-assistant.yml [hunk 6]
```diff
@@ -182,82 +220,161 @@ jobs:
...
+      command: test
+      projects: 'templates/typescript/samples/sample-skill/functional-tests/SkillSample.FunctionalTests/SkillSample.FunctionalTests.csproj'
+      arguments: '/p:CollectCoverage=true --filter TestCategory=UnitTests'
+      publishTestResults: false
 
   - pwsh: |
-     Invoke-WebRequest -Uri https://aka.ms/installazurecliwindows -OutFile .\AzureCLI.msi; Start-Process msiexec.exe -Wait -ArgumentList '/I AzureCLI.msi /quiet'
+     az login --service-principal --username "$(ServicePrincipal)" --password "$(AzureDevOps-ServicePrincipal-Secret)" --tenant "$(Azure_Tenant)"
     
+     ./deployment/scripts/deploy.ps1 -name $(VirtualAssistantBotName)  -location $(Location) -languages "en-us" -appId $(VirtualAssistantBotAppId) -appPassword $(VirtualAssistantBotAppPassword) -createLuisAuthoring -luisAuthoringRegion $(LuisAuthoringRegion) -logFolder "deployment" -luisEndpoint "https://westus.api.cognitive.microsoft.com/"
```
Comment (Batta32) [inline templates/typescript/samples/sample-assistant/functional-tests/sample-assistant.yml:345]:
```suggestion
     ./deployment/scripts/deploy.ps1 -name $(VirtualAssistantBotName) -location $(Location) -languages "en-us" -appId $(VirtualAssistantBotAppId) -appPassword $(VirtualAssistantBotAppPassword) -createLuisAuthoring -luisAuthoringRegion $(LuisAuthoringRegion) -logFolder "deployment" -luisEndpoint "https://westus.api.cognitive.microsoft.com/"
```

**Comment by DiegoCardozo94** [comment]
@Batta32 If the description it's okay, I will create the PR in MS. 👍

**Comment by Batta32** [comment]
🐑🤡

**Pull request closed without merging**